public/js/components/Expressions.js

```diff
@@ -3,70 +3,131 @@ const { connect } = require("react-redux");
 const { bindActionCreators } = require("redux");
 const ImPropTypes = require("react-immutable-proptypes");
 // const classnames = require("classnames");
+const Svg = require("./utils/Svg");
 const actions = require("../actions");
 const { getExpressions, getPause } = require("../selectors");
-// const ObjectInspector = React.createFactory(require("./ObjectInspector"));
+const Rep = React.createFactory(require("./Rep"));
 // const { truncateStr } = require("../utils/utils");
 const { DOM: dom, PropTypes } = React;
 
 require("./Expressions.css");
 
 const Expressions = React.createClass({
   propTypes: {
-    pauseInfo: ImPropTypes.map,
     expressions: ImPropTypes.list,
     addExpression: PropTypes.func,
     updateExpression: PropTypes.func,
-    evaluateExpression: PropTypes.func,
-    loadObjectProperties: PropTypes.func,
-    command: PropTypes.func
+    deleteExpression: PropTypes.func
   },
 
   displayName: "Expressions",
 
-  addExpression(e) {
-    if (e.key === "Enter") {
-      const expression = {
-        input: e.target.value
-      };
-      if (e.target.id) {
-        expression.id = e.target.id.split("-").pop();
-      }
-      this.props.addExpression(expression);
+  inputKeyPress(e, { id }) {
+    if (e.key !== "Enter") {
+      return;
     }
+    const { addExpression } = this.props;
+    const expression = {
+      input: e.target.value
+    };
+    if (id !== undefined) {
+      expression.id = id;
+    }
+    e.target.value = "";
+    addExpression(expression);
   },
 
-  updateExpression(e) {
+  updateExpression(e, { id }) {
+    e.stopPropagation();
+    const { updateExpression } = this.props;
     const expression = {
-      id: e.target.id.split("-").pop(),
-      input: e.target.textContent.split(" --> ")[0]
+      id,
+      input: e.target.textContent
     };
-    this.props.updateExpression(expression);
+    updateExpression(expression);
+  },
+
+  renderExpressionValue(value) {
+    if (!value) {
+      return;
+    }
+    if (value.exception) {
+      return Rep({ object: value.exception });
+    }
+    if (typeof value.result === "boolean") {
+      return `${value.result}`;
+    }
+    if (typeof value.result !== "object") {
+      return value.result;
+    }
+    if (typeof value.result === "object") {
+      return Rep({ object: value.result });
+    }
+  },
```

Comment:
this can be simplified to use Rep if it's an object or exception, otherwise return value.result